src/shared/components/home/emojis-form.tsx

```diff
@@ -0,0 +1,431 @@
+        let item = {
+            ... this.state!.custom_emojis[index],
+            keywords: value,
+            changed:true,
+        }
+        custom_emojis[index] = item;
+        this.setState({ custom_emojis });
+    }
+
+    handleDeleteEmojiClick(index: number, cv: CustomEmojiViewForm) {
+        if (cv.id != 0) {
+            const deleteForm: DeleteCustomEmoji = {
+                id: cv.id,
+                auth: myAuth()!
+            };
+            WebSocketService.Instance.send(wsClient.deleteCustomEmoji(deleteForm));
+        }
+        else {
+            let custom_emojis = [...this.state!.custom_emojis];
+            custom_emojis.splice(index, 1);
+            this.setState({ custom_emojis });
+        }
+    }
+
+    handleEditEmojiClick(cv: CustomEmojiViewForm) {
+        const keywords = cv.keywords.split(" ").filter(x => x.length > 0);
+        if (cv.id != 0) {
+            const editForm: EditCustomEmoji = {
+                id: cv.id,
+                category: cv.category,
+                image_url: cv.image_url,
+                alt_text: cv.alt_text,
+                keywords: keywords,
+                auth: myAuth()!
+            };
+            WebSocketService.Instance.send(wsClient.editCustomEmoji(editForm));
+        }
+        else {
+            const createForm: CreateCustomEmoji = {
+                category: cv.category,
+                shortcode: cv.shortcode,
+                image_url: cv.image_url,
+                alt_text: cv.alt_text,
+                keywords: keywords,
+                auth: myAuth()!
+            };
+            WebSocketService.Instance.send(wsClient.createCustomEmoji(createForm));
+        }
+    }
+
+    handleAddEmojiClick(event: any) {
+        event.preventDefault();
+        let custom_emojis = [...this.state!.custom_emojis];
+        let item: CustomEmojiViewForm = {
+            id: 0,
+            shortcode: "",
+            alt_text: "",
+            category: "",
+            image_url: "",
+            keywords: "",
+            changed: true,
+        }
+        custom_emojis.push(item);
+        this.setState({ custom_emojis });
+    }
+
+    async handleImageUpload(event: any, index: number) {
+        event.preventDefault();
+        let file = event.target.files[0];
+        const formData = new FormData();
+        formData.append("images[]", file);
+
+        this.setState({ loading: true });
+
+        let res: any = await fetch(pictrsUri, {
+            method: "POST",
+            body: formData,
+        });
+        let data = await res.json();
+        this.setState({ loading: false });
+        if (data.msg != "ok") {
+            toast(JSON.stringify(data), "danger");
+        }
+        else {
+            let hash = data.files[0].file;
+            let url = `${pictrsUri}/${hash}`;
+            this.handleEmojiImageUrlChange(url, index)
+        }
+    }
+
+    configurePicker(): any {
```

Comment:
In the other example, set the picker to {}, why does it need to be configured here but not there?

Reply:
The admin panel has custom picker options that remove all the built-in emojis and only shows custom emojis